### PR TITLE
Add test to check SAMPLE_COVERAGE capability with sample number from …

### PR DESCRIPTION
**Pull request by qjia7**
…1 to max sample number

The added tests will test sample coverage capbility with sample number from 1 to max sample number
which also includes the samples that may not be in the supported sample list by
getInternalformatParameter(gl.RENDERBUFFER, internalFormat, gl.SAMPLES)

### CODE REVIEW

**Comment by qjia7** [comment]
@zhenyao @kenrussell Please take a look.

File: sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    gl.blitFramebuffer(0, 0, size, size, 0, 0, size, size, gl.COLOR_BUFFER_BIT, gl.NEAREST);
+    gl.bindFramebuffer(gl.READ_FRAMEBUFFER, resolveFbo);
+
+    var pixels = new Uint8Array(size * size * 4);
+    gl.pixelStorei(gl.PACK_ALIGNMENT, 4);
+    gl.readPixels(0, 0, size, size, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "readpixels should succeed");
+
+    // Compare
+    var isOk = true;
```
Comment (qkmiao) [inline sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html]: Can you use wtu.checkCanvasRect to compare the result with expected color?

Comment (qjia7) [inline sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html]: Good suggestion. Done, thanks.

File: sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    wtu.drawFloatColorQuad(gl, quadColor);
+
+    gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, resolveFbo);
+    gl.blitFramebuffer(0, 0, size, size, 0, 0, size, size, gl.COLOR_BUFFER_BIT, gl.NEAREST);
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, resolveFbo);
+    wtu.checkCanvasRect(gl, 0, 0, size, size, [255, 0, 0, 255],
+                        "User buffer has been rendered to red with sample = "
+                        + sampleCount + ", coverageValue = " + coverageValue
+                        + " and isInverted = " + isInverted);
```
Comment (zhenyao) [inline sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html]: Can we add a tolerance of 2 or 3 here? That should still be able to verify the intended behavior.

File: sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        runTest(gl, i, false);
+        runTest(gl, i, true);
+    }
+}
+
+function runTest(gl, sampleCount, isInverted) {
+    // Setup multi-sample RBO
+    var msColorRbo = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, msColorRbo);
+    gl.renderbufferStorageMultisample(gl.RENDERBUFFER, sampleCount, gl.RGBA8, size, size);
```
Comment (zhenyao) [inline sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html]: How does Intel fail on this? It generates a GL error? If yes, then you need to check the GL error here.  If it fails in other fashion, can you describe how it fails?

File: sdk/tests/conformance2/renderbuffers/00_test_list.txt [hunk 4]
```diff
@@ -3,4 +3,5 @@ framebuffer-test.html
 framebuffer-texture-layer.html
 invalidate-framebuffer.html
 multisampled-renderbuffer-initialization.html
+multisample-with-full-sample-counts.html
```
Comment (zhenyao) [inline sdk/tests/conformance2/renderbuffers/00_test_list.txt]: Please put it to 2.0.1

File: sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html (hunk 3 above)
Comment (qjia7) [inline sdk/tests/conformance2/renderbuffers/multisample-with-full-sample-counts.html]: It's not a GL error. Intel driver can process it which is not in the supported list. But the result is not completely red when the coverage is 1. See below result:
intel_error

**Comment by qjia7** [comment]
@zhenyao  Please take another look

**Comment by qkmiao** [comment]
I tested it passed on Linux Mesa, Linux NVIDIA, Mac Intel and Mac AMD.

**Comment by zhenyao** [comment]
Thank you. LGTM

**Pull request merged**